packages/xai/src/xai-chat-settings.ts

```diff
@@ -0,0 +1,24 @@
+// https://console.x.ai and see "View models"
+export type XaiChatModelId = 'grok-beta' | (string & {});
+
+export interface XaiChatSettings {
+  /**
+Whether to enable parallel function calling during tool use. Default to true.
+   */
+  parallelToolCalls?: boolean;
+
+  /**
+A unique identifier representing your end-user, which can help OpenAI to
+monitor and detect abuse. Learn more.
+*/
+  user?: string;
```

Comment:
Do they support these options?

Reply:
- `parallelToolCalls`: not documented in their API, attempt to prompt multiple tool calls in one response not initially successful
- `user`: yes, it's noted as an optional field [here](https://docs.x.ai/api/endpoints#chat-completions)
- `downloadImages`: only helpful if model can take image data as input, which it doesn't appear to support today

Plan:
- `parallelToolCalls`: remove
- `user`: leave as-is (and add it to their provider page, I see it's on Azure's for example)
- `downloadImages`: remove